lib/ZInit.c

```diff
@@ -146,6 +151,21 @@ ZInitialize(void)
        if (hostent && hostent->h_addrtype == AF_INET)
 	   memcpy(&servaddr, hostent->h_addr, sizeof(servaddr));
 
+       // Field 10 contains our external IP.
+       mp=notice.z_message;
+       *(notice.z_message+notice.z_message_len-1) = 0;
+       for (nf=0;mp<notice.z_message+notice.z_message_len && nf<10; nf++) {
+	 mp += strlen(mp)+1;
+       }
+       if (nf==10 && mp<notice.z_message+notice.z_message_len) {
+	 inet_aton(mp, &__My_addr);
+       }
+       // Field 11 contains the IGD root URL (if UPnP is enabled)
```

Comment:
What's an IGD root URL?

Reply:
Internet Gateway Device. It's the UPnP endpoint that you have to do UDP-HTTP to... (!)